### PR TITLE
Update API review process

### PR DESCRIPTION
**Pull request by terrajobst**
Our current API review process is a bit too complicated. We want to streamline it a bit and clarify criteria for closure and who is responsible for transitions.

@weshaggard @KrzysztofCwalina @joshfree

### CODE REVIEW

File: Documentation/project-docs/api-review-process.md [hunk 1]
```diff
@@ -1,51 +1,37 @@
...
+1. **Requester files an issue**. The issue description should contain a speclet that represents a sketch of the new APIs, including samples on how the APIs are being used. The goal isn't to get a complete API list, but a good handle on how the new APIs would roughly look like and in what scenarios they are being used. Here is [a good example](https://github.com/dotnet/corefx/issues/271).
 
-* **Pull request is being reviewed**. The community reviews the code for the pull request. The review should focus on the code changes and architecture - not the APIs themselves. Once at least two project owners give their OK, the PR is considered good to go.
+2. **We assign an owner**. We'll assign a dedicated owner from our side that
```
Comment (stephentoub) [inline Documentation/project-docs/api-review-process.md]: At this point, what are the owners responsibilities?  What's the timeframe?  What are their next steps? Etc.

File: Documentation/project-docs/api-review-process.md [hunk 2]
```diff
@@ -1,51 +1,37 @@
...
 
-* **Pull is tagged as "Needs API Review"**. The project owner then marks the pull request as "Needs API Review".
+3. **Discussion**. The goal of the discussion is to help the assignee to make a
+decision whether we want to pursue the proposal or not. In this phase, the goal
+isn't necessarily to perform an in-depth review; rather, we want to make sure
+that the proposal is actionable, i.e. has a concrete design, a sketch of the
+APIs and some code samples that show how it should be used. If changes are necessary, the requester is encouraged to edit the issue description. This allows folks joining later to understand the most recent proposal. To avoid confusion, the requester should maintain a tiny change log, like a bolded "Updates:" followed by a bullet point list of the updates that were being made.
 
-* **API review**. Using the information in the pull request we'll create an APIX file that constitutes the API delta. The API review board meets multiple times a week to review all PRs that are tagged as needing an API review.
+4. **Owner makes decision**. When the owner believes enough information is available to make a decision, shell will update the issue accordingly:
```
Comment (stephentoub) [inline Documentation/project-docs/api-review-process.md]: "shell"?

File: Documentation/project-docs/api-review-process.md [hunk 3]
```diff
@@ -1,51 +1,37 @@
...
+    * **Close as won't fix**. Similarly, if proposal is taking the product in a direction we don't want to go, the issue might also get closed.
 
-* **Pull request is updated with the results of the API Review**. Once the API review is complete, the project owner uploads the notes and API HTML diff, including all comments. The project owner also updates the PR accordingly, with either a call to action to address some concerns or a good to go indicator.
+5. **API gets reviewed**. The group conducting the review is called *FXDC*, which stands for *framework design core*. In the review, we'll take notes and provide feedback. After the review, we'll publish the notes in the [API Review repository](https://github.com/dotnet/apireviews). A good example is the [review of immutable collections](https://github.com/dotnet/apireviews/tree/master/2015-01-07-immutable). Multiple outcomes are possible:
 
-* **Pull request is merged**. When there are no issues - or the issues were addressed by the contributor, the PR is merged.
+    * **Approved**. In this case the label `api-ready-for-review` is replaced
```
Comment (stephentoub) [inline Documentation/project-docs/api-review-process.md]: After Approved, then what?

**Comment by mmitche** [comment]
@dotnet-bot test this please

File: Documentation/project-docs/api-review-process.md (hunk 2 above)
Comment (terrajobst) [inline Documentation/project-docs/api-review-process.md]: she'll :-)

File: Documentation/project-docs/api-review-process.md (hunk 1 above)
Comment (terrajobst) [inline Documentation/project-docs/api-review-process.md]: Same as with all issues assigned to a person. Our goal is to get the number of pending issues to be low; I don't think we can come with a hard rule on timelines. But as explained in the process: the expectation is that the owner is responsible for:

1. Making sure the issue is actionable.
2. Making a decision whether to move the issue forward and bring it to FXDC or rejecting it.

File: Documentation/project-docs/api-review-process.md (hunk 3 above)
Comment (terrajobst) [inline Documentation/project-docs/api-review-process.md]: Same as today, we accept contributions. Do you think we should track this explicitly on the item? I don't think we ever used 'accepting PRs'.

Comment (stephentoub) [inline Documentation/project-docs/api-review-process.md]: My primary point is that someone reading this doesn't know what "same as today" means... we should just be explicit about next steps.  For example, not only are we accepting PRs, but those PRs should be against the future branch, right?  And how does someone indicate that they're working on it?  And what does it actually mean for it to be accepted, e.g. if someone submits a PR and it's merged, when does that API actually make it into some shipping version somewhere? Etc.  My concern is that there's ambiguity as to the overall process, and it'd be great if we could clear that up.

**Comment by terrajobst** [comment]
@weshaggard @stephentoub @KrzysztofCwalina I assume we're happy with the updated proposal? In that case I'll merge it and update the labels.

File: Documentation/project-docs/api-review-process.md [hunk 4]
```diff
@@ -1,51 +1,48 @@
...
+# API Review Process
 
-The .NET Framework has a long standing history of taking API usability extremely seriously. Thus, we generally review every single API that is added to the product. This page discusses how we conduct API reviews for components that are open sourced.
+The .NET Framework has a long standing history of taking API usability extremely seriously. Thus, we generally review every single API that is added to the product. This page discusses how we conduct design reviews for components that are open sourced.
 
-## Process Goals
+## Which APIs should be reviewed?
 
-The key goals are:
+The rule of thumb is that we (**corefx**) review every API that is being added to the `System` namespace. In some cases, we also review APIs that are added to namespaces, such as `Microsoft`. We mostly do this for high impact APIs, such as Roslyn, and when both the owner of the technology and us feel there is win-win for both sides if we review the APIs. However, we can't scale to review all APIs being added to .NET.
```
Comment (akoeplinger) [inline Documentation/project-docs/api-review-process.md]: Probably a stupid question, but this means `System.*` and not only `System` right? The last sentence is a bit ambiguous about what is considered ".NET".

Comment (stephentoub) [inline Documentation/project-docs/api-review-process.md]: @akoeplinger, it should probably be clarified; the intent was anything anywhere under System or any child namespace.

Nit: "we also review APIs that are added to namespaces, such as `Microsoft`" should say "that are added to other namespaces"

Nit: "and us" => "and we"

Comment (terrajobst) [inline Documentation/project-docs/api-review-process.md]: Excellent points. Fixed.

**Pull request merged**